mcs/class/corlib/System/AppDomainSetup.cs

```diff
@@ -142,9 +143,8 @@ static string GetAppBase (string appBase)
 			if (appBase == null)
 				return null;
 
-			int len = appBase.Length;
-			if (len >= 8 && appBase.ToLower ().StartsWith ("file://")) {
-				appBase = appBase.Substring (7);
+			if (appBase.ToLower (CultureInfo.InvariantCulture).StartsWith ("file://")) {
```

Comment:
This would be better:
```suggestion
			if (appBase.StartsWith ("file://", StringComparison.OrdinalIgnoreCase)) {
```